src/mem/Pool.cc

```diff
@@ -259,7 +246,7 @@ memPoolGetGlobalStats(MemPoolGlobalStats * stats)
 
     /* gather all stats for Totals */
     iter = memPoolIterate();
-    while ((pool = memPoolIterateNext(iter))) {
+    while (auto *pool = memPoolIterateNext(iter)) {
```

Comment:
If we are going to polish this code, let's improve its const-correctness:

```suggestion
    while (const auto pool = memPoolIterateNext(iter)) {
```

I do not insist on this change. There are many other PR-modified places where const-correctness can be improved. I am not going to mark any of them, but, if you do not mind, please improve them. Also, if you prefer that I do these changes, please let me know, and I will.

Reply:
Unfortunately the getStats() logic alters the pool. For now I would like to to leave those deep logic changes out of scope.
Adding a TODO now.

Reply:
TODO added in [b48d53b](https://github.com/squid-cache/squid/pull/1181/commits/b48d53b23db6f2b01f52b8b94f7a63e7e6a31ea9)

Reply:
You are misinterpreting what the suggestion does. It is unrelated to the TODO you have added. Try applying the change. If it works, consider improving other, similar changed places as well.

Reply:
I unresolved this old change request because it was not resolved.

Reply:
```
../../../src/mem/Pool.cc: In function 'int memPoolGetGlobalStats(MemPoolGlobalStats*)':
../../../src/mem/Pool.cc:238:27: error: passing 'const MemImplementingAllocator' as 'this' argument discards qualifiers [-fpermissive]
  238 |         if (pool->getStats(&pp_stats, 1) > 0)
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
In file included from ../../../src/mem/Pool.h:31,
                 from ../../../src/mem/PoolChunked.h:12,
                 from ../../../src/mem/Pool.cc:14:
../../../src/mem/Allocator.h:30:17: note:   in call to 'virtual int Mem::Allocator::getStats(MemPoolStats*, int)'
   30 |     virtual int getStats(MemPoolStats *, int accumulate = 0) = 0;
      |                 ^~~~~~~~

```

Reply:
Please try my suggestion (instead of trying something I am _not_ suggesting). My suggestion does not make MemImplementingAllocator constant. Moreover, I just tried it (using official Squid code), and it does work as expected.

Reply:
Oh, its the asterisk. sorry.